### ISSUE TITLE
[Bug] Unity 2021.2 gradle fails to resolve local maven path

### ISSUE THREAD

**Issue by toxikman** [googlesamples/unity-jar-resolver#478]
### [REQUIRED] Please fill in the following fields:

  * Unity editor version: Unity 2021.2.3 (Silicon Mac)
  * External Dependency Manager version: 1.2.167
  * Source you installed EDM4U: UPM tgz file
  * Features in External Dependency Manager in use: Android Resolver, iOS Resolver, VersionHandler
  * Plugins SDK in use: Google Play Games + others (see [mainTemplate.gradle](https://github.com/googlesamples/unity-jar-resolver/files/7574984/mainTemplate.gradle.txt))
  * Platform you are using the Unity editor on: Macbook Pro M1 Max, MacOS Monterey 12.0.1

### [REQUIRED] Please describe the issue here:
Gradle reports an error while looking for package "com.google.games:gpgs-plugin-support:0.10.13" from the Google Play Games services plugin:

    Could not find com.google.games:gpgs-plugin-support:0.10.13.
    Searched in the following locations:
    file:/Assets/Other/LocalMavenRepo/GooglePlayGames/Editor/m2repository/com/google/games/gpgs-plugin-support/0.10.13/gpgs-plugin-support-0.10.13.pom

The package is stored in a local maven repository. This works in Unity 2020.3.23 LTS, but not in 2021.2.3. The error shows the path is not fully qualified so I suspect the code that EDM4U adds to mainTemplate.gradle isn't working as expected:

    url (unityProjectPath + "/Assets..."

Please see attached [mainTemplate.gradle](https://github.com/googlesamples/unity-jar-resolver/files/7574984/mainTemplate.gradle.txt) file that is being used.

#### Please answer the following, if applicable:
100% Repro

Also see possible related issue: https://github.com/googlesamples/unity-jar-resolver/issues/477

**Comment by deetmo** [comment]
I can confirm this issue. 
Unity 2021.2.3 will resolve DIR_UNITYPROJECT variable to empty string. 
This results in 100% failed builds when any android dependency is required. 

Don’t know if it works in 2021.2.4.

**Comment by deetmo** [comment]
For those who want to make it working…
Here is my hax script to do it. 
Place it in some Editor folder. 

```using UnityEditor;
using UnityEditor.Android;
using UnityEngine;
using System.IO;

class GradleBuildHax : IPostGenerateGradleAndroidProject
{
	private const string ToReplace = "def unityProjectPath = $/file:////$.replace(\"\\\\\", \"/\")";
	
    public int callbackOrder { get { return 0; } }
    public void OnPostGenerateGradleAndroidProject(string path)
    {
        Debug.Log("GradleBuildHax at path " + path);
		
		var projectPath = Path.GetFullPath(Path.Combine(Application.dataPath, "..")).Replace("\\", "/");
		
		Debug.Log("GradleBuildHax projectPath " + projectPath);
        
        var gradlePath = Path.Combine(path, "build.gradle");
		
		var targetValue = "def unityProjectPath = \"file:///" + projectPath + "\"";
        
        var text = File.ReadAllText(gradlePath);
        text = text.Replace(ToReplace, targetValue);
        File.WriteAllText(gradlePath, text);
		
		Debug.Log("GradleBuildHax gradle:\n" + text);
    }
}

**Comment by chkuang-g** [comment]
@deetmo Good find. So the missing `DIR_UNITYPROJECT` is the culprit here.

I think this Firebase issue is caused by this as well, since `firebase-app-unity` is from a local Maven repo as well.
https://github.com/firebase/quickstart-unity/issues/1205

Let me ping the team to get a patch for this.

**Comment by chkuang-g** [comment]
This seems to be a regression from the Unity side

https://issuetracker.unity3d.com/issues/android-template-variable-star-star-dir-unityproject-star-star-is-not-properly-substituted-when-exporting-a-gradle-project?_ga=2.122711496.294034660.1640030138-1623586750.1636598340

Their fix has been integrated to `2022.1.0a13` but it is still in-review for `2021.2`.

This is...rather inconvenience.

**Comment by chkuang-g** [comment]
Thanks for the note!  I will close this bug for now.  Please report a new one if the problem still occurs.
